src/main/java/net/snowflake/client/core/SFSession.java

```diff
@@ -535,6 +536,10 @@ public synchronized void open() throws SFException, SnowflakeSQLException {
               getWarehouse()));
     }
 
+    // Initialize QCC
+    int qccSize = loginOutput.getQueryContextCacheSize();
+    qcc = new QueryContextCache(qccSize);
```

Comment:
it does not seem possible for this QueryContext cache to be null, so I would make it final in the declaration and remove all the null checks since it would be guaranteed by the Java compiler.

Reply:
I check the Chen's server side changes and it mentions that it can be null. We can discuss again and make changes in the driver accordingly.

Reply:
Yes, I can see that on the server side it can be null but here in the client code, we always allocate it regardless of the size, no?

Reply:
Removed redundant null checks, as those places session must be open. We can't make it final as size can be configure from server. That value is coming as part of open session.